src/DotNet.Glob/Evaluation/AnyCharacterTokenEvaluator.cs

```diff
@@ -12,10 +12,12 @@ public AnyCharacterTokenEvaluator(AnyCharacterToken token)
             _token = token;
         }
 
+        public bool IsMatch(string allChars, int currentPosition, out int newPosition)
```

Comment:
There is certainly no performance advantage to having an API that takes a `string` as an argument.
Here's why I'm leaving them out.

In order to disable implicit conversions, the caller must hold `ReadOnlySpan<char>`.
Like this:.
```
var span = "path/somefile.txt";
glob.IsMatch(span);
span = span.Slice(5);
glob.IsMatch(span);
```
If the caller doesn't have a ReadOnlySpan, then the implicit conversion from `string` to `ReadOnlySpan<char>` already happens every time.
So the number of conversions required should be the same, even if there is an overload that takes a `string` as an argument.

Implicit conversions are also marked `[MethodImpl(MethodImplOptions.AggressiveInlining)]`.
It is optimized by JIT and is lightly loaded, as [ASM only requires 5 instructions.](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHuGuf6GASwB2GBgFkAFMXIAGBrgCUHPj34B2BgGUADtmFSlAblUMAvqbWmBIsbv1SASjGwATAPLCANgE97wgB4wAAtsKAA+BWVOajU1Yk1cADoASWFXGAR3ADNJAHIAKjzFE1ieCzLuGjMgA)

The other reason to keep the API for passing strings is the compatibility of the code that uses this project.
In a rare case, migrating from `net4*` to `netcoreapp*` results in an error for the following code
```
Func<string, bool> isMatchDelegate = glob.IsMatch;
```
This is a benefit to users of this library.

Reply:
> So the number of conversions required should be the same, even if there is an overload that takes a string as an argument.

But that kind of makes the overload that takes a string, redundant, as you said yourself, if the caller holds a ReadOnlySpan then they will use that method, if they hold a string, then an implicit conversion would take place, so having an overload that takes a string and does the conversion seems unnecessary in both those circumstances.. However I know you have another point about compatibility..

Reply:
> The other reason to keep the API for passing strings is the compatibility of the code that uses this project.
In a rare case, migrating from net4* to netcoreapp* results in an error for the following code

Its a good point.. and well considered. I think keeping the api surface as compatible as possible across different platforms, and making Span based API an "addition" to the api rather than a substitution is the better way to go in general so I think we are in agreement. Thanks for taking the time to discuss.